platform/viewer/src/lib/DownloadViewportEngine.js

```diff
@@ -0,0 +1,137 @@
+const DOWNLOAD_ELEMENT_ID = 'download-element';
+const PREVIEW_ELEMENT_ID = 'preview-element';
+
+class DownloadViewportEngine {
+  constructor() {
+    this.$previewElement = null;
+    this.$downloadElement = document.createElement('div');
+    this.$activeViewport = null;
+    this.reRenderActionCache = null;
+    this.showAnnotations = false;
+
+    // TODO - Maybe there is a way to get all these tools dynamically?
+    this.availableTools = [
+      'LengthTool',
+      'WwwcTool',
+      'BidirectionalTool',
+      'AngleTool',
+      'StackScrollTool',
+      'BrushTool',
+      'FreehandMouseTool',
+      'EllipticalRoiTool',
+      'CircleRoiTool',
+      'RectangleRoiTool'
+    ];
+
+    this.updateHash = null;
+
+    this.mountPreview = this.mountPreview.bind(this);
+    this.save = this.save.bind(this);
+    this.setElementSize = this.setElementSize.bind(this);
+    this.toggleAnnotations = this.toggleAnnotations.bind(this);
+    this.updateCache = this.updateCache.bind(this);
+    this.enableCornerstoneTools = this.enableCornerstoneTools.bind(this);
+    this.showPreview = this.showPreview.bind(this);
+  }
+
+  setElementSize(element, prop, value) {
+    const resizingElement = element ? element : this.$downloadElement;
+    resizingElement.style[prop] = `${value}px`;
+    const canvas = resizingElement.querySelector('canvas');
+    canvas.style[prop] = `${value}px`;
+  }
+
+  toggleAnnotations() {
+    const element = this.$downloadElement;
+    this.availableTools.forEach(tool => {
+      if (this.showAnnotations) {
+        cornerstoneTools.setToolEnabledForElement(element, tool.replace('Tool', ''));
+      } else {
+        cornerstoneTools.setToolDisabledForElement(element, tool.replace('Tool', ''));
+      }
+    });
+  }
+
+  updateCache(
+    previewElemReference = this.$previewElement,
+    activeViewport =  this.$activeViewport,
+    reRenderAction = this.reRenderActionCache,
+    showAnnotations = this.showAnnotations,
+  ){
+    this.$previewElement = previewElemReference;
+    this.$activeViewport = activeViewport;
+    this.reRenderActionCache = reRenderAction;
+    this.showAnnotations = showAnnotations;
+  }
+
+  enableCornerstoneTools(element = this.$downloadElement) {
```

Comment:
UI components should never touch `cornerstone` or `cornerstone-tools` API

Reply:
platform/viewer/src/lib/DownloadViewportEngine.js  is an Engine, not the component itself. Maybe its a misunderstood? Can we  keep these communications with cornerstone or cornerstone-tools here?